### PR TITLE
`Integrated code lifecycle`: Fix Git authentication if the password contains a colon

### PR DESCRIPTION
**Pull request by Hialus**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

@bensofficial discovered an issue when trying to authorize with LocalVC. This was caused, because his password contained a colon (`:`).

### Description

The cause of this was that the credentials were parsed by LocalVC using `String::split(":")`, which would break if the password contained colons as well.
This PR fixes this by instead using `String::indexOf(":")` in combination with `String::substring`.

### Steps for Testing

Prerequisites:
- 1 Account with a colon in the password on a test server with LocalVC

1. Clone an exercise
2. Try to authorize using your password
3. With this PR it should work, without it won't

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

Unchanged

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved the handling of credentials in the authentication process to ensure correct parsing and secure login.
	- Updated authentication method in `LocalVCServletService.java` for more reliable extraction of username and password.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/localvc/LocalVCServletService.java [hunk 1]
```diff
@@ -205,12 +205,13 @@ private User authenticateUser(String authorizationHeader) throws LocalVCAuthExce
 
         String basicAuthCredentials = checkAuthorizationHeader(authorizationHeader);
 
-        if (basicAuthCredentials.split(":").length != 2) {
+        if (basicAuthCredentials.split(":").length < 2) {
```
[the remark this answers is not in the document]
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/localvc/LocalVCServletService.java:208]: This is how it should work. There should be at least one colon in the credentials.

[the remark this answers is not in the document]
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/localvc/LocalVCServletService.java]: I improved it further. Much better now

**Review by Strohgelaender — APPROVED** [review]
changes make sense. Can you add a server test for this case?

**Review by bensofficial — APPROVED** [review]
Thanks. Great that my (former) password is now working when cloning repos.

**Review by MichaelOwenDyer — APPROVED** [review]
Fantastic 💯

**Review by bassner — APPROVED** [review]
We'll add a server test later. Need this fix asap for the upcoming migration

**Pull request merged**